CBinding/CLang/CLangManager.cs

```diff
@@ -0,0 +1,482 @@
+﻿using System;
+using CBinding;
+using ClangSharp;
+using MonoDevelop.Ide;
+using MonoDevelop.Projects;
+using System.Collections.Generic;
+using System.Threading;
+using MonoDevelop.Ide.Editor;
+using System.Runtime.InteropServices;
+using CBinding.Refactoring;
+using CBinding.Parser;
+using System.IO;
+
+namespace CBinding
+{
+	/// <summary>
+	/// Class to manage clang Translation units, thread safe
+	/// </summary>
+	public class CLangManager : IDisposable
+	{
+		public readonly object syncroot = new object ();
+		private CProject project;
+		private CXIndex index;
+		private Dictionary<string, CXTranslationUnit> translationUnits;
+
+		public Dictionary<string, CXTranslationUnit> TranslationUnits {
+			get {				
+				lock (syncroot) {
```

Comment:
This lock is redundant - getting the field is atomic. You should probably add a comment about how the dictionary itself isn't threadsafe and accesses to the dictionary need to be locked on the SyncRoot. Or ideally hide this entirely behind threadsafe API, or use a ConcurrentDictionary? Or maybe switch to locking the dictionary, instead of a syncroot object?